lib/ecto/changeset.ex

```diff
@@ -1372,15 +1375,20 @@ defmodule Ecto.Changeset do
       error_param = "#{param}_confirmation"
       error_field = String.to_atom(error_param)
       value = Map.get(changeset.params, param)
-
       case Map.fetch(changeset.params, error_param) do
         {:ok, ^value} -> []
         {:ok, _}      -> [{error_field, {message(opts, "does not match confirmation"), []}}]
-        :error        -> []
+        :error        -> confirmation_missing(opts, error_field)
       end
     end
   end
 
+  defp confirmation_missing(opts = [required: true], error_field) do
```

Comment:
You can't pattern match on keyword like that. What if I pass `[message: "foo", required: true]` as options?